### PR TITLE
chore(gatsby-link): Minor refactor

### PR DESCRIPTION
**Pull request by jamesgeorge007**
## Description
Minor refactor to have a more concise version.

## Related Issues
N/A

### CODE REVIEW

File: packages/gatsby-link/src/index.js [hunk 1]
```diff
@@ -178,6 +178,11 @@ GatsbyLink.propTypes = {
   replace: PropTypes.bool,
 }
 
+const showWarningMessage = action =>
```
Comment (sidharthachatterjee) [inline packages/gatsby-link/src/index.js]: Want to call the function showDeprecationWarning and accept functionName, altFunctionName and version?

**Comment by jamesgeorge007** [comment]
cc @sidharthachatterjee

File: packages/gatsby-link/src/index.js (hunk 1 above)
Comment (sidharthachatterjee) [reply packages/gatsby-link/src/index.js]: Would've preferred to take `altFunctionName` and `version` too as parameters too (so we could reuse this function) but this is valuable as well so thank you!

**Comment by jamesgeorge007** [comment]
@sidharthachatterjee should I shoot up yet another PR with the changes as you suggested above :question:

**Comment by sidharthachatterjee** [comment]
@jamesgeorge007 Sure! That would be awesome 🥇

**Comment by jamesgeorge007** [comment]
@sidharthachatterjee would also love to know your thoughts on [Mevn-CLI](https://github.com/madlabsinc/mevn-cli). Consider, leaving a :star:

**Pull request merged**